content/articles/cross-site-request-forgery-attack/index.md

```diff
@@ -0,0 +1,126 @@
+
+`http://bankexample.com/onlinetransfer?amount=$5000&account=attacker`
+
+The attacker then tricks the user into clicking the above malicious URL. He uses social engineering mechanisms to force the user into executing the URL. The malicious URL is embedded into HTML content and sent to the user.
+
+`<a href="http://bankexample.com/onlinetransfer?amount=$50000&account=attacker">Click here now!</a>`
+
+Once the user clicks the link, funds are transferred to the attacker's account without his consent. However, if the victim has an active session with the online banking application, the request is treated to be legit.
+
+#### POST requests
+If the online banking application uses the HTTP POST request method to transfer funds, sending malicious requests using links is impossible.
+
+However, the malicious request can be submitted using a form. Also, the attacker can use JavaScript to designate and enable the form to be executed automatically.
+
+Below is how the form may look like:
+
+```JavaScript
+body onload="document.forms[0].submit()">
+   <form action=" http://bankexample.com/onlinetransfer" method="POST">
+     <input type="hidden" name="account" value="attacker"/>
+     <input type="hidden" name="amount" value="$50000"/>
+     <input type="submit" value="Click here now!"/>
+   </form>
+ </body>
+```
+
+### Impacts of CSRF attacks
+CSRF is a dangerous vulnerability that abuses the trust between the victim's browser and the web server. The magnitude of its impact depends on the number of benefits allocated to the user.
+
+For instance, if the victim is a normal user, the attacker can fully control the user's account. However, if the victim has an administrative role, the attacker might exploit the entire web application.
+
+CSRF has been used to perform several malicious activities such as stealing data, unauthorized funds transfer, changing passwords, damaging clients' relationships, spreading worms or malware, and many more.
+
+### Testing of CSRF attacks
+CSRF testing is the process of finding CSRF vulnerabilities in web applications. Testing of CSRF vulnerabilities can be done either manually or using automated tools. 
+
+#### Manual testing for CSRF attacks
+If you want to find out if the session is not secure, you need to examine the web app's session. If session handling is on the client-side and displays data to the browser, the web app is vulnerable. The data displayed on the browser is the HTTP authentication credentials and cookies.
+
+Resources accessible via the HTTP GET requests are no doubt vulnerable. Even though JavaScript automates POST requests, they are exposed and vulnerable. Therefore, the use of POST only hardly solves CSRF vulnerabilities.
+
+The tester can use the following methods of testing:
+- Black box testing
+- Gray box testing
+
```

Comment:
Please provide explanation about these testing terms or add links to pages that explain more about them.